From 7c782da4518edbc4e6ae5a8ca8ee319f362af437 Mon Sep 17 00:00:00 2001
Message-ID: <7c782da4518edbc4e6ae5a8ca8ee319f362af437.1752333776.git.sam@gentoo.org>
From: Sam James <sam@gentoo.org>
Date: Sat, 12 Jul 2025 16:22:02 +0100
Subject: [PATCH] ipa-modref: don't call flags_from_decl_or_type on
 non-inlined_to node

	PR ipa/120987

gcc/ChangeLog:

	* ipa-modref.cc (ipa_merge_modref_summary_after_inlining): Drop
	duplicate, unguarded flags_from_decl_or_type call.
---
 gcc/ipa-modref.cc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/gcc/ipa-modref.cc b/gcc/ipa-modref.cc
index d3e510195b53..cbf5042d14bf 100644
--- a/gcc/ipa-modref.cc
+++ b/gcc/ipa-modref.cc
@@ -5347,7 +5347,6 @@ ipa_merge_modref_summary_after_inlining (cgraph_edge *edge)
   cgraph_node *n;
   for (n = edge->caller; n->inlined_to; n = n->callers->caller)
     flags |= flags_from_decl_or_type (n->decl);
-  flags |= flags_from_decl_or_type (n->decl);
   bool ignore_stores = ignore_stores_p (edge->caller->decl, flags);
 
   if (!callee_info && to_info)

base-commit: e6d3c88e7bb07f94308074f9751e4384a191e022
-- 
2.50.1

